ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -157,7 +154,6 @@ public void operationComplete(ChannelFuture future) throws Exception {
                       http2ClientResponseHandler.getResponseInfoQueue()
                           .put(new ResponseInfo(requestInfoFromChannelAttr, NetworkClientErrorCode.NetworkError, null));
                     }
-                    ReferenceCountUtil.safeRelease(requestInfo.getRequest());
```

Comment:
Discussed this offline. This was removed b/c it was found that writeAndFlush will release the buffer in both the success and failure case.

Reply:
just want to confirm this statement: "writeAndFlush will release the buffer in both the success and failure case" without any exception?